#!/usr/bin/env cwl-runner
cwlVersion: v1.0
class: CommandLineTool
doc: string
requirements:
  DockerRequirement:
    dockerPull: quay.io/biocontainers/canu:1.8--pl526h470a237_0
baseCommand: canu

  ~/ERR2092776.fastq -p canu genomeSize=100m maxThreads=4

inputs:
  nanopore_raw:
    type: boolean
    label: "reads were generated by nanopore technology"
    default: true
    inputBinding:
      prefix: -nanopore-raw
  assembly_directory:
    type: string
    label: "assembly directory"
    default: "canu"
    inputBinding:
      prefix: -d
  output_prefix:
    type: string
    label: "assembly prefix"
    default: "canu"
    inputBinding:
      prefix: -p
  genomeSize:
    type: string
    label: "best guess of the haploid genome size"
    doc: "The genome size should be your best guess of the haploid genome size of what is being assembled.  It is used primarily to estimate coverage in reads, NOT as the desired assembly size.  Fractional values are allowed: '4.7m' equals '4700k' equals '4700000'"
    inputBinding:
      prefix: genomeSize=
      separate: false
  maxThreads:
    type: int
    label: "maxThreads"
    default: 4
    inputBinding:
      prefix: maxThreads=
      separate: false
  input_fastq:
    type: File
    label: "input fastq reads"
    inputBinding:
      position: 1

outputs:
  contigs:
    type: File
    outputBinding:
      glob: "$(inputs.assembly_directory)/$(inputs.output_prefix).contigs.fasta"
  stdout: stdout
  stderr: stderr
stdout: canu-stdout.log
stderr: canu-stderr.log
